site/_posts/2023-03-09-react-example-application.md

```diff
@@ -0,0 +1,247 @@
+
+The login / register view in FusionAuth can be themed using the FusionAuth theme editor. You can find more information about the theme editor in the [FusionAuth documentation](https://fusionauth.io/docs/v1/tech/themes/).
+
+For this example application, we integrate Tailwind CSS and DaisyUI into the theme by automatically generating the css stylesheet based on the FusionAuth theme templates.
+
+## Benefits SDK
+
+The React SDK enables developers to implement the login, logout, and registration flows in a React application.
+
+It provides pre-built buttons for login, logout, and registration. It also provides a `withFusionAuth` [HOC](https://reactjs.org/docs/higher-order-components.html) that can be used to wrap a component and provide the user information to the component.
+With `RequireAuth` you can wrap a component and require the user to be authenticated before the component is rendered. The `withRole` option also enables you to restrict the user to a specific role.
+
+`RequireAuth` is used in the `LoggedInMenu` component, which is used to display the user information, profile, and logout button if the user is authenticated.
+
+With the `FusionAuthConfig` you can configure the FusionAuth instance that is used by the React SDK. If you are using a different OAuth 2.0 backend, you will most likely have to configure the route props.
+
+We are using the React Router v6 in the example application. To protect routes, we are using a custom `ProtectedRoutes` component. This component uses the `useFusionAuth` hook to check if the user is authenticated. If the user is authenticated, the component renders the `Outlet` component. If the user is not authenticated, the component redirects the user to the `/` route.
+
+```tsx
+export const ProtectedRoutes: FC = () => {
+  const {isLoading, isAuthenticated} = useFusionAuth();
+
+  if (isLoading) return null;
+
+  return (isAuthenticated ? <Outlet/> : <Navigate to={'/'} replace={true}/>);
+};
+```
+
+{% include _callout-note.liquid
+content=
+"For more information about the React SDK, see the [FusionAuth React SDK](/docs/v1/tech/client-libraries/react-sdk) documentation."
+%}
+
+{% include _callout-note.liquid
+content=
+"Any code snippets in this article are excerpts from the example application. You can find the full source code on GitHub."
+%}
+
+### Login Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-login.png" alt="Login / Register Prompt if the user is not authenticated" class="img-fluid" figure=false %}
+
+The login page allows the user to login or register using the FusionAuth login / register flow. The login page is implemented in the `LoginPage.tsx` file and displays the `login` and `register` buttons.
+We are not using the pre-built button components from the React SDK. Instead, we are using the `useFusionAuth` hook to get the `login` and `register` methods from the `FusionAuthContext`. This allows us to customize the login / register button.
```

Comment:
`We are not using the pre-built button components from the React SDK.` - fine, but please explain why.